blog/2023/04/julia-1.9-highlights.md

```diff
@@ -0,0 +1,392 @@
+
+
+### `pkg> add` will only auto update the registry once per day
+
+*Kristoffer Carlsson*
+
+Pkg will now remember the last time the registry was updated across julia sessions and only auto-update once per day when using an `add` command. Previously the registry auto-updated once per session. Note that the `update` command
+
+### `pkg> add` can now try to only add already installed packages
+
+*Ian Butterworth*
+
+When working with many environments, for instance across Pluto notebooks, the default behavior of `Pkg.add` to add the latest version of the requested package and any new dependencies can mean hitting precompilation frequently.
+
+`Pkg.add` can now be told to prefer to add already installed versions of packages, which are more likely to be precompiled.
+
+To globally opt-in to the new preference set the env var `JULIA_PKG_PRESERVE_TIERED_INSTALLED` to `true`.
+
+Or to enable for specific operations use:
+- `pkg> add --preserve=tiered_installed Foo` to try this new strategy first in the tiered preserve.
+- `pkg> add --preserve=installed Foo` to strictly try this strategy, or error.
+
+### `pkg> why` to tell you why a package in the manifest
+
+*Kristoffer Carlsson*
+
+To show the reason why a package is in the manifest a new `pkg> why Foo` command is available. The output is all the different ways to reach the package through the dependency graph starting from the dependencies.
+
+```
+(jl_zMxmBY) pkg> why DataAPI
+  CSV → PooledArrays → DataAPI
+  CSV → Tables → DataAPI
+  CSV → WeakRefStrings → DataAPI
+```
+
+### `pkg> test --coverage` (the default for CI) is now faster
+
+*Ian Butterworth*
+
+Previously, coverage testing could only be enabled for either `all` where all code visited is checked, `user` (the prior default for `Pkg.test`) where everything except for `Base` is checked including stdlibs, or `none` where tracking is disabled.
+
+The github action `julia-runtest` defaults to coverage testing on, meaning a lot of tracking outside of the package under test previously took place, slowing tests down, especially tight loops.
+
+v1.8 introduced the ability to specify a path to either a file or directory for coverage to be tracked in via `--code-coverage=@path`, and v1.9 makes that the default for `Pkg.test(coverage=true)` (and thus used by `julia-runtest` by default).
+
+This change means often a lot less code needs to be tracked, and in cases where code from dependencies fall in tight loops this can heavily speed up the test suite. In one example Octavian.jl tests with coverage enabled went from >2hrs to ~6 minutes.
+
+## LLVM Update to v14
```

Comment:
@vchuravy @gbaraldi are there more highlights from llvm 14 that could be added here?

Reply:
@pchintalapudi is this the first version with IRCE?

Reply:
No, IRCE was introduced in 1.8.

Reply:
Didn't SVE work better on LLVM14 than it did on 13 @giordano ?